eng/targetingpacks.targets

```diff
@@ -0,0 +1,110 @@
+    <EnableTargetingPackDownload>false</EnableTargetingPackDownload>
+  </PropertyGroup>
+
+  <!-- .NETCoreApp 2.x DisableImplicitAssemblyReferences support. -->
+  <Choose>
+    <When Condition="'$(DisableImplicitAssemblyReferences)' == 'true' and
+                     '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                     $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '3.0'))">
+    <PropertyGroup>
+      <MicrosoftNetCoreAppRefPackRefDir>$(PkgMicrosoft_NETCore_App)\ref\$(_ShortFrameworkIdentifier)$(_ShortFrameworkVersion)\</MicrosoftNetCoreAppRefPackRefDir>
+    </PropertyGroup>
+    <ItemGroup>
+      <PackageReference Update="Microsoft.NETCore.App"
+                        ExcludeAssets="all"
+                        GeneratePathProperty="true" />
+    </ItemGroup>
+    </When>
+  </Choose>
+
+  <!-- Simple name references will be resolved from the targeting pack folders and should never be copied to the output. -->
+  <ItemGroup>
+    <Reference Update="@(Reference)">
+      <Private Condition="'%(Reference.Extension)' != '.dll'">false</Private>
+    </Reference>
+  </ItemGroup>
+
+  <!-- .NETCoreApp > 2.x DisableImplicitAssemblyReferences support. -->
+  <Target Name="RemoveFrameworkReferences"
+          BeforeTargets="_HandlePackageFileConflicts"
+          AfterTargets="ResolveTargetingPackAssets"
+          Condition="'$(DisableImplicitAssemblyReferences)' == 'true' and
+                     '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                     $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '3.0'))">
+    <PropertyGroup>
+      <MicrosoftNetCoreAppRefPackRefDir>%(ResolvedFrameworkReference.TargetingPackPath)\ref\$(_ShortFrameworkIdentifier)$(_ShortFrameworkVersion)\</MicrosoftNetCoreAppRefPackRefDir>
+    </PropertyGroup>
+    <ItemGroup>
+      <Reference Remove="@(Reference)"
+                 Condition="'%(Reference.FrameworkReferenceName)' == '$(SharedFrameworkName)'" />
+    </ItemGroup>
+  </Target>
+
+  <!-- Add the resolved targeting pack to the assembly search path. -->
+  <Target Name="UseTargetingPackForAssemblySearchPaths"
+          BeforeTargets="ResolveAssemblyReferences;
+                         DesignTimeResolveAssemblyReferences"
+          Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
+    <PropertyGroup>
+      <AssemblySearchPaths>$(AssemblySearchPaths);$(MicrosoftNetCoreAppRefPackRefDir.TrimEnd('/\'))</AssemblySearchPaths>
+      <DesignTimeAssemblySearchPaths>$(DesignTimeAssemblySearchPaths);$(MicrosoftNetCoreAppRefPackRefDir.TrimEnd('/\'))</DesignTimeAssemblySearchPaths>
+    </PropertyGroup>
+  </Target>
+
+  <!-- Use local targeting pack for NetCoreAppCurrent. -->
+  <Target Name="UpdateTargetingAndRuntimePack"
+          Condition="'$(_UseLocalTargetingRuntimePack)' == 'true'"
+          AfterTargets="ResolveFrameworkReferences">
+    <ItemGroup>
+      <ResolvedTargetingPack Update="@(ResolvedTargetingPack)"
```

Comment:
I believe `ResolvedTargetingPack Update="@(ResolvedTargetingPack)"` is redundant.  In a target if you omit include/update/remove you are operating on all items.

Reply:
@rainersigwald just to double check, is this indeed true?

Reply:
Docs : https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-items?view=vs-2019#updating-metadata-on-items-in-an-itemgroup-of-a-target

Reply:
I see, so this can only be omitted inside a Target. Thanks